src/utils/api-request.ts

```diff
@@ -50,6 +51,43 @@ export class HttpRequestHandler {
       data?: Object,
       headers?: Object,
       timeout?: number): Promise<Object> {
+    // Convenience for calling the real _sendRequest() method with the original params.
+    const sendOneRequest = () => {
+      return this._sendRequest(host, port, path, httpMethod, data, headers, timeout);
+    };
+
+    return sendOneRequest()
+      .catch((response: { statusCode: number, error: string|Object }) => {
+        // Retry if the request failed due to a network error.
+        if (response.error instanceof FirebaseAppError) {
+          if ((<FirebaseAppError> response.error).hasCode(AppErrorCodes.NETWORK_ERROR)) {
```

Comment:
By the way, all request errors are currently considered network errors:
`req.on('error', (error) => {...})`
That could be too general. Would it be better to catch ECONNRESET errors and retry only in that case?

Reply:
Indeed this retries for any non-HTTP and non-timeout error: connection reset, broken pipe, etc.  @inlined You're most familiar with the ways we've seen this present, do you think it's worth passing up the original error and having a whitelist here?

Reply:
ECONNRESET is the one off the top of my head.